javaslang/src/test/java/javaslang/control/TryTest.java

```diff
@@ -247,6 +253,268 @@ public void shouldThrowNullPointerExceptionWhenCallingTryRunRunnable() {
+        assertThat(resource5.isClosed).isTrue();
+        assertThat(resource6.isClosed).isTrue();
+    }
+
+    @Test
+    public void shouldCreateFailureTryWithResources6() {
+        final Resource<Integer> resource1 = resource(1);
+        final Resource<Integer> resource2 = resource(2);
+        final Resource<Integer> resource3 = resource(3);
+        final Resource<Integer> resource4 = resource(4);
+        final Resource<Integer> resource5 = resource(5);
+        final Resource<Integer> resource6 = resource(6);
+        final Try<?> actual = Try.withResources(() -> resource1, () -> resource2, () -> resource3, () -> resource4, () -> resource5, () -> resource6).of((i1, i2, i3, i4, i5, i6) -> { throw new Error(); });
+        assertThat(actual.isFailure()).isTrue();
+        assertThat(resource1.isClosed).isTrue();
+        assertThat(resource2.isClosed).isTrue();
+        assertThat(resource3.isClosed).isTrue();
+        assertThat(resource4.isClosed).isTrue();
+        assertThat(resource5.isClosed).isTrue();
+        assertThat(resource6.isClosed).isTrue();
+    }
+
+    @Test
+    public void shouldCreateSuccessTryWithResources7() {
+        final Resource<Integer> resource1 = resource(1);
+        final Resource<Integer> resource2 = resource(2);
+        final Resource<Integer> resource3 = resource(3);
+        final Resource<Integer> resource4 = resource(4);
+        final Resource<Integer> resource5 = resource(5);
+        final Resource<Integer> resource6 = resource(6);
+        final Resource<Integer> resource7 = resource(7);
+        final Try<String> actual = Try.withResources(() -> resource1, () -> resource2, () -> resource3, () -> resource4, () -> resource5, () -> resource6, () -> resource7).of((i1, i2, i3, i4, i5, i6, i7) -> "" + i1.value + i2.value + i3.value + i4.value + i5.value + i6.value + i7.value);
+        assertThat(actual).isEqualTo(Success("1234567"));
+        assertThat(resource1.isClosed).isTrue();
+        assertThat(resource2.isClosed).isTrue();
+        assertThat(resource3.isClosed).isTrue();
+        assertThat(resource4.isClosed).isTrue();
+        assertThat(resource5.isClosed).isTrue();
+        assertThat(resource6.isClosed).isTrue();
+        assertThat(resource7.isClosed).isTrue();
+    }
+
+    @Test
+    public void shouldCreateFailureTryWithResources7() {
+        final Resource<Integer> resource1 = resource(1);
+        final Resource<Integer> resource2 = resource(2);
+        final Resource<Integer> resource3 = resource(3);
+        final Resource<Integer> resource4 = resource(4);
+        final Resource<Integer> resource5 = resource(5);
+        final Resource<Integer> resource6 = resource(6);
+        final Resource<Integer> resource7 = resource(7);
+        final Try<?> actual = Try.withResources(() -> resource1, () -> resource2, () -> resource3, () -> resource4, () -> resource5, () -> resource6, () -> resource7).of((i1, i2, i3, i4, i5, i6, i7) -> { throw new Error(); });
```

Comment:
for full coverage you may want to test this with `InterruptedException` or some other non-fatal exception.
Please see the coverage of `Try` after this commit for further details :)